### PR TITLE
Fix support for master.key with final newline

### PR DESCRIPTION
**Pull request by mfcollins3**
When using Tuist with encrypted code signing certificates in GitHub Actions and storing the private encryption key in a repository secret, GitHub Actions is appending a newline to the end of the master key. This causes Tuist to be unable to decrypt the code signing certificates because it is using the newline as part of the private key. This change modifies the `SigningCipher.masterKey(at:)` function to read the master.key file as a text file, trim the newlines, and then to get the raw bytes for the characters to be hashed into the encryption key.

Resolves https://github.com/tuist/tuist/issues/4781

### Short description 📝

The `master.key` file was being read as a binary blob. GitHub Actions is appending a newline to the end of the encryption key when it is stored as a secret. This change reads `master.key` as a text file, trims the newline characters, and then obtains the bytes of the private key.

### How to test the changes locally 🧐

I created the `SigningCipherTests.test_decrypt_signing_when_master_key_has_newline()` test case to illustrate and validate this change.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] The title of the PR will be used as changelog entry, please make sure it is clear and suitable
- [x] In case the PR introduces changes that affect users, the documentation has been updated
- [ ] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`

### CODE REVIEW

File: Sources/TuistSigning/SigningCipher.swift [hunk 1]
```diff
@@ -215,7 +215,9 @@ public final class SigningCipher: SigningCiphering {
         else { throw SigningCipherError.signingDirectoryNotFound(path) }
         let masterKeyFile = rootDirectory.appending(components: Constants.tuistDirectoryName, Constants.masterKey)
         guard FileHandler.shared.exists(masterKeyFile) else { throw SigningCipherError.masterKeyNotFound(masterKeyFile) }
-        let plainMasterKey = try FileHandler.shared.readFile(masterKeyFile)
+        let plainMasterKey = try FileHandler.shared.readTextFile(masterKeyFile)
+            .trimmingCharacters(in: .newlines)
+            .data(using: .utf8)!
         return plainMasterKey.sha256()
```
[the remark this answers is not in the document]
Comment (mfcollins3) [reply Sources/TuistSigning/SigningCipher.swift:221]: It probably could be, but it’s your code and nothing else was using `readMasterKey`, so I figured there was some reason not to use it. I can change it tomorrow if you would like me to.

[the remark this answers is not in the document]
Comment (danyf90) [reply Sources/TuistSigning/SigningCipher.swift:221]: Sure! feel free to change it 💪

**Comment by danyf90** [comment]
@allcontributors add @mfcollins3 for code

File: Sources/TuistSigning/SigningCipher.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfcollins3) [reply Sources/TuistSigning/SigningCipher.swift]: I fixed the implementation. Thank you.

**Review by danyf90 — APPROVED** [review]

**Pull request merged**